zoo_project.py

```diff
@@ -34,35 +35,55 @@ class Cage:
     """
     cage_counter = 1
 
+    def add_predator(self, new_animal):
+        """Adds a predator to the cage
+            When a predator it's added to the cage the existing animals
+            are checked against the list of preys, removing the ones that match
+            as they are eaten by the new predator.
+        """
+        for animal in self.animals:
+            if animal.__class__ in new_animal.preys:
+                self.delete_animal(animal)
+                print(("{} just ate {}, this happened because you are placing predators \
+                       in the same cages as their preys!")
+                      .format(new_animal.name, animal.name))
+        self.animals.append(new_animal)
+
+    def add_prey(self, new_animal):
```

Comment:
Let's say you have an animal who eats rats, but is prey to lions.  How will your algorithm cope with animals who are potentially both prey and predators?

I suggest adjusting your animals and then writing a test first, before adjusting anything.